src/appleseed-max-impl/appleseedvolumemtl/appleseedvolumemtl.cpp

```diff
@@ -0,0 +1,593 @@
+}
+
+IOResult AppleseedVolumeMtl::Load(ILoad* iload)
+{
+    IOResult result = IO_OK;
+
+    while (true)
+    {
+        result = iload->OpenChunk();
+        if (result == IO_END)
+            return IO_OK;
+        if (result != IO_OK)
+            break;
+
+        switch (iload->CurChunkID())
+        {
+          case ChunkFileFormatVersion:
+            {
+                USHORT version;
+                result = read<USHORT>(iload, &version);
+            }
+            break;
+
+          case ChunkMtlBase:
+            result = MtlBase::Load(iload);
+            break;
+        }
+
+        if (result != IO_OK)
+            break;
+
+        result = iload->CloseChunk();
+        if (result != IO_OK)
+            break;
+    }
+
+    return result;
+}
+
+Color AppleseedVolumeMtl::GetAmbient(int mtlNum, BOOL backFace)
+{
+    return Color(0.0f, 0.0f, 0.0f);
+}
+
+Color AppleseedVolumeMtl::GetDiffuse(int mtlNum, BOOL backFace)
+{
+    Color color;
+    m_pblock->GetValue(ParamIdVolumeScattering, GetCOREInterface()->GetTime(), color, FOREVER);
+    return color;
+}
+
+Color AppleseedVolumeMtl::GetSpecular(int mtlNum, BOOL backFace)
+{
+    return Color(0.0f, 0.0f, 0.0f);
+}
+
+float AppleseedVolumeMtl::GetShininess(int mtlNum, BOOL backFace)
+{
+    return 0.0f;
+}
+
+float AppleseedVolumeMtl::GetShinStr(int mtlNum, BOOL backFace)
+{
+    return 0.0f;
+}
+
+float AppleseedVolumeMtl::GetXParency(int mtlNum, BOOL backFace)
+{
+    float multiplier;
+    m_pblock->GetValue(ParamIdVolumeScatteringMultiplier, GetCOREInterface()->GetTime(), multiplier, FOREVER);
+    return asf::min(1 - (multiplier / 100.0f), 0.95f);
+}
+
+void AppleseedVolumeMtl::SetAmbient(Color c, TimeValue t)
+{
+}
+
+void AppleseedVolumeMtl::SetDiffuse(Color c, TimeValue t)
+{
+    m_pblock->SetValue(ParamIdVolumeScattering, t, c);
+}
+
+void AppleseedVolumeMtl::SetSpecular(Color c, TimeValue t)
+{
+}
+
+void AppleseedVolumeMtl::SetShininess(float v, TimeValue t)
+{
+}
+
+void AppleseedVolumeMtl::Shade(ShadeContext& sc)
+{
+}
+
+int AppleseedVolumeMtl::get_sides() const
+{
+    return asr::ObjectInstance::BothSides;
+}
+
+bool AppleseedVolumeMtl::can_emit_light() const
+{
+    return false;
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedVolumeMtl::create_material(
+    asr::Assembly&  assembly,
+    const char*     name,
+    const bool      use_max_procedural_maps)
+{
+    asr::ParamArray material_params;
+    auto time = GetCOREInterface()->GetTime();
+
+    //
+    // Volume.
+    //
+
+    asr::ParamArray volume_params;
+
+    // Absorption.
+    Color absorption;
+    m_pblock->GetValue(ParamIdVolumeAbsorption, time, absorption, m_params_validity);
+    const auto absorption_color_name = std::string(name) + "_volume_absorption";
+    insert_color(assembly, absorption, absorption_color_name.c_str());
+    volume_params.insert("absorption", absorption_color_name);
+
+    // Absorption Multiplier.
+    float absorption_multiplier = 0.0f;
```

Comment:
Do we need to initialize these variables?

Reply:
Actually since we are dropping max 2015 I can use betteer versions, like `GetColor`, `GetFloat`, etc.